### PR TITLE
Addon Vitest: Support modifying mergeConfig on addon setup

### PR DESCRIPTION
**Pull request by yannbf**
Relates to #32291

## What I did

This PR improves two things:
1. The detection of valid vitest config files during Storybook init
2. The modification of vitest config files in the vitest addon postinstall

It now supports a [commonly used](https://github.com/search?q=%22export%20default%20mergeConfig(%22%20path%3Avitest.config.*&type=code) pattern like so:
```ts
import { defineConfig, mergeConfig } from 'vitest/config';
import viteBaseConfig from './vite.base.config';
import path from 'path';

export default mergeConfig(
  viteBaseConfig,
  defineConfig({
    test: {
       // ...
    },
  })
);
```

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=<PR_NUMBER>`_

## Release Notes

* **New Features**
  * Better support for Vitest configs that use merge-style composition, including multi-argument cases.
  * Improved merging behavior to consolidate test/project/workspace entries into a consistent structure.

* **Tests**
  * Expanded test coverage across many Vitest config shapes and edge cases to ensure safer config updates.
  * Added scenarios validating non-modification for unsupported or unchanged configurations.

### CODE REVIEW

File: code/addons/vitest/src/updateVitestFile.test.ts [hunk 1]
```diff
@@ -422,6 +422,203 @@ describe('updateConfigFile', () => {
...
+            environment: 'jsdom',
+          }
+        })
+      )
+    `);
+
+    const updated = updateConfigFile(source, target);
+    expect(updated).toBe(true);
+
+    const { code } = babel.generate(target);
```
Comment (ndelangen) [inline code/addons/vitest/src/updateVitestFile.test.ts]: Do you want to run prettier over this before passing it into the toMatchInlineSnapshot for easier reviewing?

File: code/addons/vitest/src/updateVitestFile.test.ts [hunk 2]
```diff
@@ -422,6 +422,203 @@ describe('updateConfigFile', () => {
...
+      import { storybookTest } from '@storybook/addon-vitest/vitest-plugin';
+      const dirname = typeof __dirname !== 'undefined' ? __dirname : path.dirname(fileURLToPath(import.meta.url));
+
+      // More info at: https://storybook.js.org/docs/next/writing-tests/integrations/vitest-addon
+      export default mergeConfig(viteConfig, defineConfig({
+        test: {
+          projects: [{
+            extends: true,
+            test: {
+              globals: true
```
Comment (ndelangen) [inline code/addons/vitest/src/updateVitestFile.test.ts]: I guess this not having a name, is okay?

Comment (yannbf) [reply code/addons/vitest/src/updateVitestFile.test.ts]: Yes, it's on purpose. I thought it wouldn't make sense to write a name on user's behalf, and leaving with no name is fine (I tested this quite a bit), Vitest will generate index based names in that case.

**Review by ndelangen — APPROVED** [review]
